docs/news/posts/2024-10-04-data-access.md

```diff
@@ -0,0 +1,283 @@
+│     │  └── inputs
+│     └── 02-iteration_01-PwCalculation
+│        ├── inputs
+│        ├── node_inputs
+│        │  └── pseudos
+│        │     └── Si
+│        └── outputs
+├── 02-seekpath-seekpath_structure_analysis
+│  └── inputs
+├── 03-scf-PwBaseWorkChain
+│  ├── ...
+...
+```
+
+</td>
+</tr>
+
+  </tbody>
+</table>
+
+Therefore, after running the command once, you'll have all data involved in the execution of your workflow directly
+accessible, and can explore it with the Unix command line utilities we all know and love [^5].
+
+So happy grepping!
+
+## New QueryBuilder Syntax
+
+_SQL queries, but intuitive!_
+
+In addition to accessing raw files as outlined above, AiiDA's powerful SQL database allows querying for stored data
+`Node`s, which can be achieved with the `QueryBuilder` (as documented
+[here](https://aiida.readthedocs.io/projects/aiida-core/en/latest/howto/query.html)). While using the `QueryBuilder` is
+(at least for most of us) easier than writing raw SQL queries, its syntax usually requires some familiarization [^6].
+
+Recent improvements have therefore enabled an alternative, more intuitive way to construct queries. Let us explain with
+the following example: Assume you wanted to obtain all integers with `pk`s in a range between 1 and 10 (both excluded)
+from a `Group` called "integers". You'd have to construct the rather convoluted query as shown in the left column of the
+following table:
+
+<!-- prettier-ignore -->
+<!-- Taken from: https://gist.github.com/panoply/176101828af8393adc821e49578ac588 -->
+<table style="table-layout: fixed; width: 100%">
+  <thead>
+    <tr>
+      <th width="500px"> Old `QueryBuilder` syntax </th>
+      <th width="500px"> New `QueryBuilder` syntax </th>
+    </tr>
+  </thead>
+  <tbody>
+  <tr width="600px">
+      <td style="word-break:break-all;">
+
+```python
+QueryBuilder().append(
+    Group,
+    filters={
+        "label": "integers",
+    },
+    project=["label"],
+    tag="group",
+).append(
+    Int,
+    with_incoming="group",
+    filters={
+        "and": [
+            {"pk": {">": 1}},
+            {"pk": {"<": 10}},
+        ]
+    },
+    project=["pk", "attributes.value"],
+)
+```
+
+</td>
+<td>
+
+```python
+QueryBuilder().append(
+    Group,
+    filters=Group.fields.label == "integers",
+    project=[Group.fields.label],
+    tag="group",
+).append(
+    Int,
+    with_incoming="group",
+    filters=(Int.fields.pk > 1) & (Int.fields.pk < 10),
+    project=[Int.fields.pk, Int.fields.value],
+)
+
+
+
+
+
+
+
+```
+
+</td>
+</tr>
+
+  </tbody>
+</table>
+
+With the new syntax, `Node` attributes are directly accessible via
+`.fields` and logic can be applied to them directly. Through this, for example the filter on the values of
+the `pk`s reduces from:
+
+```python
+filters={
+    "and": [
+        {"pk": {">": 1}},
+        {"pk": {"<": 10}},
+    ]
+},
+```
+
+to the more concise:
+
+```python
+filters=(Int.fields.pk > 1) & (Int.fields.pk < 10),
```

Comment:
maybe
```suggestion
filters=(Int.fields.pk > 1) & (Int.fields.pk < 10)
```

Reply:
That's what happens when I don't ruff-autoformat the shit out of any code snippet I write :D